### ISSUE TITLE
Reconfigure UI Shell stories to show fully accessible examples

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13419]
Surfaced in #12323:

> ### Side rail implementation on Storybook failing to handle content in breakpoint - and has an 'impossible' version.
> 
> The non-persistent left nav that arises from responsive design is quite similar to the Side rail variation, which reveals a consistently non-persistent left nav. It's quite a nice implementation, and Carbon may want to consider how the variations can be merged into something consistent and best.
> 
> However, the implementation of the [side rail with header](https://react.carbondesignsystem.com/iframe.html?id=components-ui-shell--side-nav-rail-w-header&viewMode=story) has a significant glitch and some cautions.
> 
> The first is, what is the point of having a side rail and a hamburger menu? They both do the same thing (and have the same content!). If the side fail is implemented, there does not seem to be any reason to have a hamburger menu.
> 
> Secondly, in the storybook implementation, the 4 links displayed at the top of the nav, which disappear due to response design at about 175% zoom, do NOT appear in the left 'rail' or in the hamburger menu. They should.
> 
> Thirdly, there is an implementation of [side rail _without_ a UI Shell header](https://react.carbondesignsystem.com/iframe.html?args=&id=components-ui-shell--side-nav-rail&viewMode=story). I believe it is impossible to have no UI Shell header -- that it is a requirement of all Carbon pages. Is there some other use case where this makes sense? 
> That aside, if you consider how the side rail works without the need for the hamburger menu here, you can see that it serves as further evidence that there is NO NEED for the hamburger menu when side rail is used. As mentioned above, where responsive design collapses the top nav, those links can just appear at the top of the side rail content

There isn't an inherent defect here, other than the stories showing components in isolation that therefore have accessibility issues. The suggested outcome of this issue is to focus towards reconfiguring the stories to match the following:

```[tasklist]
- [ ] Only show fully baked, in-flight, and accessible examples in the uishell stories (ones that include a header and all related pieces of UI necessary to be compliant with our guidance)
- [ ] Isolated stories should be removed ~or put behind a dev-only flag #12971~
```
